sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_response_handlers.py

```diff
@@ -27,10 +27,28 @@
     TextAnalyticsWarning
 )
 
+def _get_too_many_documents_error(obj):
+    try:
+        too_many_documents_errors = [
+            error for error in obj.errors if error.id == ""
+        ]
+    except AttributeError:
+        too_many_documents_errors = [
+            error for error in obj["errors"] if error["id"] == ""
+        ]
+    if too_many_documents_errors:
+        return too_many_documents_errors[0]
+    return None
+
 class CSODataV4Format(ODataV4Format):
 
     def __init__(self, odata_error):
+
         try:
+            if not odata_error.get("error"):
+                odata_error = _get_too_many_documents_error(odata_error)
+            if not odata_error:
+                raise ValueError("The object given does not include an error.")
```

Comment:
if i read this as a user, it makes me think the object I gave did not include an error. Maybe change to something like "Service encountered an error without any details"?

Reply:
yep sounds good. apologies for rejecting your stale review now :'(